pkgs/tools/admin/vncdo/default.nix

```
@@ -1,26 +1,26 @@
 { stdenv, fetchFromGitHub
-, python2Packages
+, pythonPackages, isPy27
 }:
-python2Packages.buildPythonPackage {
+pythonPackages.buildPythonPackage rec {
   pname = "vncdo";
-  version = "0.11.2";
+  version = "0.12.0";
 
   src = fetchFromGitHub {
     owner = "sibson";
     repo = "vncdotool";
-    rev = "5c03a82dcb5a3bd9e8f741f8a8d0c1ce082f2834";
-    sha256 = "0k03b09ipsz8vp362x7sx7z68mxgqw9qzvkii2f8j9vx2y79rjsh";
+    rev = "v${version}";
+    sha256 = "0h3ccr8zi7xpgn6hz43x1045x5l4bhha7py8x00g8bv6gaqlbwxn";
   };
 
-  propagatedBuildInputs = with python2Packages; [
+  propagatedBuildInputs = with pythonPackages; [
     pillow
     twisted
     pexpect
     nose
     ptyprocess
   ];
 
-  doCheck = false;
+  doCheck = !isPy27;
```

```suggestion
  doCheck = true;
```

python3Packages are not available under Python 2.7

---

Yes, but we can still call this package with e.g. `python2Packages.vncdo`
```
vncdo/python2-and-3-support [$] ❯ nix build -f . python2Packages.vncdo -L
python2.7-vncdo> Sourcing python-remove-tests-dir-hook
python2.7-vncdo> Sourcing python-catch-conflicts-hook.sh
python2.7-vncdo> Sourcing python-remove-bin-bytecode-hook.sh
```

---

We actually can't because this mixes python 2 and python 3. Please remove that functionality or do not pass in pythonPackages.

---

Not sure where the mixing is happening, but I'm fine with specifying the Python dependencies explicitly.
The hashes did not seem to change after I made the change, but now the expression is at least marginally cleaner.

---

> Please remove that functionality or do not pass in pythonPackages.

No, do *not* pass in individual Python libraries and functions in the top-level. We pass in the package set for consistency.

---

There is more, I'll continue in the main thread.